index.js

```diff
@@ -0,0 +1,145 @@
+const todoList = document.getElementById('todo-list');
+let todoListStorage = [];
+
+const filterList = ({ target }) => {
+  let newTodoList = [];
+  let todoItems = [...todoList.children];
+
+  if (todoListStorage.length) {
+    todoItems = todoListStorage;
+  }
+
+  if (target.className.includes('all')) {
+    newTodoList = todoItems
+      .filter((item) => item)
+      .map((item) => `<li>${item.innerHTML}</li>`);
+  }
+  if (target.className === 'active') {
+    newTodoList = todoItems
+      .filter((item) => !item.children[0].children[0].checked)
+      .map((item) => `<li>${item.innerHTML}</li>`);
+  }
+  if (target.className === 'completed') {
+    newTodoList = todoItems
+      .filter((item) => item.children[0].children[0].checked)
+      .map((item) => `<li>${item.innerHTML}</li>`);
+  }
+
+  todoList.innerHTML = newTodoList;
+  todoList.innerHTML = todoList.innerHTML.replaceAll(',', '');
+  document.getElementsByClassName(
+    'todo-count'
+  )[0].children[0].innerHTML = parseInt(todoList.children.length, 10);
+};
+
+const filterAll = document.getElementsByClassName('all')[0];
+const filterTodo = document.getElementsByClassName('active')[0];
+const filterCompleted = document.getElementsByClassName('completed')[0];
+filterAll.addEventListener('click', filterList);
+filterTodo.addEventListener('click', filterList);
+filterCompleted.addEventListener('click', filterList);
+
+const handleCheckBox = ({ target }) => {
+  const li = target.parentNode.parentNode;
+  if (target.checked) {
+    target.checked = true;
+    li.classList.add('completed');
+  } else {
+    target.checked = false;
+    li.classList.remove('completed');
+  }
+};
+
+const deleteItem = ({ target }) => {
+  const li = target.parentNode.parentNode;
+  li.parentNode.removeChild(li);
+  count = document.getElementsByClassName('todo-count')[0].children[0]
+    .innerHTML;
+  document.getElementsByClassName('todo-count')[0].children[0].innerHTML =
+    parseInt(count, 10) - 1;
+  todoListStorage = [...todoList.children];
+};
+
+const showHideDeleteButton = (e) => {
+  if (e.target.tagName !== 'DIV') {
+    return;
+  }
+  if (e.type === 'mouseover') {
+    e.target.children[2].style.visibility = 'visible';
+  }
+
+  if (e.type === 'mouseleave') {
+    e.target.children[2].style.visibility = 'hidden';
+  }
+};
+
+const showEditInput = ({ target }) => {
+  target.parentNode.children[3].value = target.innerHTML;
+  target.parentNode.children[3].style.display = 'inline-block';
+  target.parentNode.parentNode.classList.add('editing');
+  target.parentNode.children[3].focus();
+};
+
+const editContent = (e) => {
+  const value = e.target.value;
+  if (e.key === 'Escape') {
+    e.target.style.display = 'none';
+    if (value === '') {
+      e.target.parentNode.parentNode.classList.remove('editing');
+      return;
+    }
+    e.target.parentNode.children[1].innerHTML = e.target.value;
+    e.target.parentNode.parentNode.classList.remove('editing');
+  }
```

Comment:
```suggestion
if (e.key !== 'Escape') {
 return
}
e.target.style.display = 'none';
if (value === '') {
  e.target.parentNode.parentNode.classList.remove('editing');
  return;
}
  e.target.parentNode.children[1].innerHTML = e.target.value;
  e.target.parentNode.parentNode.classList.remove('editing');
}
```

위와 같이 작성하면 depth도 줄이고 더 코드를 이해하기 쉬워질 것 같아요~! 가능하면 early return을 통해 코드를 만들면 depth가 대부분 줄어든답니다 😄